tests/integration/query.ts

```diff
@@ -50,3 +58,130 @@ it('universal autocomplete sets search intents', async () => {
   expect(statefulCore.state.query.searchIntents).toEqual(['NEAR_ME']);
 });
 
+it('vertical autocomplete get latest request results', async () => {
+  const mockedCore: any = {
+    verticalAutocomplete: jest.fn(
+      async (_request: VerticalAutocompleteRequest) => {
+        const waitTime = _request.input?.length * 200;
+        return new Promise(res => setTimeout(() => res(
+          { results: [ {value: _request.input} ]}), waitTime));
+      })
+  };
+  const stateManager = new ReduxStateManager();
+  const statefulCore = new StatefulCore(mockedCore, stateManager);
+  const queries = ['long request', 'short'];
+  statefulCore.setVerticalKey('someKey');
+
+  const dispatchEventSpy = jest.spyOn(stateManager, 'dispatchEvent');
+  statefulCore.setQuery(queries[0]);
+  statefulCore.executeVerticalAutoComplete();
+  statefulCore.setQuery(queries[1]);
+  statefulCore.executeVerticalAutoComplete();
+
+  await new Promise(res => setTimeout(res, 3000));
```

Comment:
instead of waiting an arbitrary time could we just wait for a `Promise.all` of the 2 autocompletes?